elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/SQLQueryEngine.java

```diff
@@ -166,7 +168,7 @@ protected SQLQuery toSQL(Query query, SQLSchema schema) {
 
         builder.projectionClause(extractProjection(query));
 
-        Set<Path.PathElement> joinPredicates = new HashSet<>();
+        Set<Path.PathElement> joinPredicates = new LinkedHashSet<>();
```

Comment:
The joins need to happen following the order in the `Path`.
Change to `LinkedHashSet` to preserve the join order.